source/MRMesh/MRMeshBuilder.h

```diff
@@ -43,15 +44,15 @@ struct VertDuplication
 
 // resolve non-manifold vertices by creating duplicate vertices
 // return number of duplicated vertices
-MRMESH_API size_t duplicateNonManifoldVertices( std::vector<Triangle>& tris,
+MRMESH_API size_t duplicateNonManifoldVertices( Triangulation & t, FaceBitSet * region = nullptr,
     std::vector<VertDuplication>* dups = nullptr );
 
 // construct mesh topology from a set of triangles with given ids;
 // unlike simple fromTriangles() it tries to resolve non-manifold vertices by creating duplicate vertices
 MRMESH_API MeshTopology fromTrianglesDuplicatingNonManifoldVertices( 
-    std::vector<Triangle> & tris,
+    Triangulation & t,
     std::vector<VertDuplication> * dups = nullptr,
-    std::vector<Triangle> * skippedTris = nullptr );
+    const BuildSettings & settings = {} );
```

Comment:
Should Trianglulation be const? (we have settings with region, to show if some triangles have not been added)?